packages/calypso-build/webpack/minify.js

```diff
@@ -117,6 +119,13 @@ module.exports = ( options ) => {
 		),
 		...terserOptions,
 	};
+	cssMinimizerOptions = {
+		preset: 'default',
+		...cssMinimizerOptions,
+	};
 
-	return [ new TerserPlugin( { ...options, terserOptions } ) ];
+	return [
+		new TerserPlugin( { parallel, terserOptions } ),
```

Comment:
Any chance one of the consumers of `calypso-build` could be providing additional options to the terser plugin? If that's possible then we might be omitting those by just leaving `parallel` there.

Reply:
Potentially, yes. But I think exposing those options to consumers will make the API harder to use.

These are all the options supported by Terser and CssMinimizer:

```
CssMinimizerPlugin({
	test
	include
	exclude
	parallel
	minify
	minimizerOptions {
		processorOptions
	}
	warningsFilter
})
TerserPlugin({
	test
	include
	exclude
	parallel
	minify
	terserOptions
	extractComments
})
```

Before this PR, consumers of `calypso-build` could specify any Terser option. The problem is that the new plugin API overlaps, so there are options (`test`, `include`, `exclude`, `parallel` and `minify`) that can potentially collide. `parallel` is a bit special in the sense that most likely consumers will like to pass `true` to both plugins, but that may not be true for the other options.

One potential solution (the approach I took in this PR) is to limit the exposed options to `minify`, `terserOptions` and `minimizerOptions`, based on my guess that those are the options people would like to customize the most.

An alternative would be to accept an object with two keys `{terserOptions, cssMinimizerOptions}` with the options for each plugin. But Terser already uses `terserOptions`, and chances are that consumers of `calypso-build` were already using it, making the transition to the new API a bit harder. Also, I'd argue that if the consumer of this package really needs to customize most of those options, then maybe they'll be better by just importing TerserPlugin/CssMinimizerPlugin directly, without `calypso-build`.

To be honest, I'm not sure what's the best approach here. What do you think?

Reply:
At first, I was leaning towards this:

> An alternative would be to accept an object with two keys {terserOptions, cssMinimizerOptions} with the options for each plugin. 

but I agree with this very much, too:

> if the consumer of this package really needs to customize most of those options, then maybe they'll be better by just importing TerserPlugin/CssMinimizerPlugin directly,

This also aligns with YAGNI, which is a guiding principle of mine 😉  As long as we don't need to customize any of the other options, I think we can go with just `parallel` for now. We can iterate and expand the API as needed in the future. One step at a time 🦶 

That being said, I'm fine with leaving it as you suggested in this PR.